internal/checker/services.go

```diff
@@ -0,0 +1,427 @@
+package checker
+
```

Comment:
This file has a bunch of checker functions that are meant to be used by services.
There are existing functions in `checker.go` that could probably move here later, e.g. `GetSymbolAtLocation`.